src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/TimeServedExternalRecordsReasonServiceTest.kt

```diff
@@ -2,118 +2,172 @@ package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.reset
 import org.mockito.kotlin.times
 import org.mockito.kotlin.verify
 import org.mockito.kotlin.whenever
 import org.springframework.security.core.Authentication
 import org.springframework.security.core.context.SecurityContext
 import org.springframework.security.core.context.SecurityContextHolder
-import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.TimeServedExternalRecords
-import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.request.TimeServedExternalRecordsRequest
-import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.request.UpdateNomisLicenceReasonRequest
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.AuditEvent
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.TimeServedExternalRecord
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.request.ExternalTimeServedRecordRequest
 import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.AuditEventRepository
 import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.StaffRepository
 import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.TimeServedExternalRecordsRepository
 import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.TestData.communityOffenderManager
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.TestData.prisonUser
+
+private const val BOOKING_ID = 12345L
+private const val PRISON_NUMBER = "A1234BC"
+private const val USERNAME = "tcom1"
+private const val REQUEST_PRISON_CODE = "MDI"
+private const val REQUEST_REASON = "Some reason"
+
+private const val PREVIOUS_PRISON_CODE = "LEI"
+private const val PREVIOUS_REASON = "old reason"
 
 class TimeServedExternalRecordsReasonServiceTest {
 
   private val licenceRepository = mock<TimeServedExternalRecordsRepository>()
   private val staffRepository = mock<StaffRepository>()
   private val auditEventRepository = mock<AuditEventRepository>()
 
-  private val service = TimeServedExternalRecordsService(
+  private val service = TimeServedExternalRecordService(
     licenceRepository,
     staffRepository,
     auditEventRepository,
   )
 
   @BeforeEach
   fun reset() {
-    org.mockito.kotlin.reset(
+    reset(
       licenceRepository,
       staffRepository,
       auditEventRepository,
     )
-    val authentication = org.mockito.kotlin.mock<Authentication>()
-    val securityContext = org.mockito.kotlin.mock<SecurityContext>()
+    val authentication = mock<Authentication>()
+    val securityContext = mock<SecurityContext>()
 
-    whenever(authentication.name).thenReturn(com.username)
+    whenever(authentication.name).thenReturn(USERNAME)
     whenever(securityContext.authentication).thenReturn(authentication)
     SecurityContextHolder.setContext(securityContext)
   }
 
-  @Test
-  fun `recordTimeServedExternalRecord throws ValidationException if staff username not found`() {
-    // Arrange
```

Comment:
There's quite a few examples of Given / When / Then  and I don't think we should start introducing Arrange / Act / Assert